apps/studio/src/server/modules/folder/folder.router.ts

```diff
@@ -12,13 +13,85 @@ export const folderRouter = router({
     }),
   readFolderOrTopLevelFolder: protectedProcedure
     .input(readFolderOrTopLevelFolderSchema)
-    .query(async ({ input, ctx }) => {
-      // TODO: Fill these in later
-      const folderName: string = ''
-      const children: { id: string; name: string; type: 'page' | 'folder' }[] =
-        []
-      // Not sure if a backpointer is needed here
-      const parentId: string = ''
+    .query(async ({ ctx, input }) => {
+      // Things that aren't working yet:
+      // 0. Perm checking
+      // 1. Last Edited user and time
+      // 2. Page status(draft, published)
+      if (input.resourceId) {
+        // Non-root level lookup
+        const folderResult = await ctx.db
+          .selectFrom('Resource')
+          .select(['name', 'parentId'])
+          .where('id', '=', input.resourceId)
+          .executeTakeFirstOrThrow()
+        const childrenResult = await ctx.db
+          .selectFrom('Resource')
+          .select(['id', 'name', 'blobId'])
+          .where('parentId', '=', input.resourceId)
+          .execute()
+        const children = childrenResult.map((c) => {
+          if (c.blobId) {
+            return {
+              id: c.id,
+              name: c.name,
+              type: 'page',
+              lastEditDate: new Date(0),
+              lastEditUser: 'Coming Soon',
+              permalink: '/placeholder',
+              status: 'published',
+            }
+          }
+          return {
+            id: c.id,
+            name: c.name,
+            type: 'folder',
+            lastEditDate: 'folder',
+            lastEditUser: 'Coming Soon',
+            permalink: '/placeholder',
+            status: 'folder',
+          }
+        })
+
+        const { parentId } = folderResult
+        const folderName = folderResult.name
+
+        return {
+          folderName,
+          children,
+          parentId,
+        }
+      }
+      const childrenResult = await ctx.db
```

Comment:
i'm abit confused here, why would `resourceId` ever be `null` and in that case, shouldn't we throw an error since this implies that they've tried to fetch a non-existing resource (`resourceId` here maps to `Resource.id`, which is the pkey, no?)

Reply:
That I believe was for the case where you only have the site id, and want to get everything in the root level 'folder',i.e. orphan pages and folders. From the discussion on the schema if i understood correctly we wanted one endpoint to get both root 'folders' as well as actual folder contents. If that's not the case let me know and I am happy to break it into 2 cases!